### PR TITLE
chore(FacetedSearch): Add documentation

### PR DESCRIPTION
**Pull request by nicomaligne**
**What is the problem this PR is trying to solve?**
This doc has been initialized on ui-dataset and at the beginning of the faceted project.
So this is not up to date.
**What is the chosen solution to this problem?**
Make the doc great again
**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/faceted-search/src/components/README.md [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+
+| Props                  | Type                              | Info                                                                    |
+| ---------------------- | --------------------------------- | ----------------------------------------------------------------------- |
+| badgesDefinitions      | badgesFacetedPropTypes            | payload of badges definition                                            |
+| badgesFaceted          | { badges: badgesFacetedPropTypes} | available badges                                                        |
+| customBadgesDictionary | object                            | a dictionary of new badges                                              |
+| initialFilterValue     | string                            | filter value at first render of the [AddFacetPopover](#AddFacetPopover) |
+| onSubmit               | func                              | callback trigger when badges change                                     |
+| setFacetedBadges       | func                              | callback trigger to sync the badges at every state change               |
+
```
Comment (vbocquet) [inline packages/faceted-search/src/components/README.md]:
```suggestion
| customOperatorsDictionary               | operatorsPropTypes                              | a dictionary  of custom operators                                     |

```

Comment (vbocquet) [reply packages/faceted-search/src/components/README.md]: Just add this props that missing

File: packages/faceted-search/src/components/README.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+Use childrenAsProps pattern to pass the current display mode.
+
+| Props       | Type   | Info                                                |
+| ----------- | ------ | --------------------------------------------------- |
+| children    | func   | render props with current faceted mode as parameter |
+| error       | string | error returns by query call                         |
+| id          | string | id shared to all children                           |
+| facetedMode | string | mode display at first render                        |
+| inProgress  | bool   | indicates query progress                            |
+
```
Comment (vbocquet) [inline packages/faceted-search/src/components/README.md]:
```suggestion
| setFacetedMode  | func   | function to set the faceted mode if it's a controlled component                            |
```

Comment (vbocquet) [reply packages/faceted-search/src/components/README.md]: just another missing props

File: packages/faceted-search/src/components/README.md [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+---
+
+### FacetedSearchIcon
+
+Simple **button icon** switch.
+
+| Props   | Type     | Info                            |
+| ------- | -------- | ------------------------------- |
+| active  | boolean  | icon is activated or not        |
+| loading | string   | display skeleton if true        |
```
Comment (vbocquet) [inline packages/faceted-search/src/components/README.md]:
```suggestion
| loading | boolean   | display skeleton if true        |
```

File: packages/faceted-search/src/components/README.md [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+| t                     | func               | i18n translate function               |
+
+### BadgeText
+
+A badge with a **text area**.
+
+| Default Operators |
+| ----------------- |
+| contains          |
+| endWidth          |
```
Comment (vbocquet) [inline packages/faceted-search/src/components/README.md]: not sure about this ones: `endWidth`, `notContains`, `regExp`, `startWidth`, where they comes from?

File: packages/faceted-search/README.md [hunk 5]
```diff
@@ -1 +1,93 @@
-## Faceted search
\ No newline at end of file
+# Faceted search
+
+## Installation
+
+Add npm package to app
```
Comment (romainseb) [inline packages/faceted-search/README.md]:
```suggestion
Add npm package to add
```
?

**Pull request merged**